catalog/harvest/catalog-harvest-common/src/main/java/org/codice/ddf/catalog/harvest/common/FileSystemPersistenceProvider.java

```diff
@@ -54,20 +55,22 @@
 
   private static final String PERSISTED_FILE_SUFFIX_REGEX = "\\.ser";
 
+  private String persistencePath;
+
   private String mapName = "default";
 
   private FilenameFilter filter;
 
-  public FileSystemPersistenceProvider() {}
-
   public FileSystemPersistenceProvider(String mapName) {
-    LOGGER.trace("INSIDE: FileSystemPersistenceProvider constructor,  mapName = {}", mapName);
+    this(mapName, "data");
+  }
+
+  public FileSystemPersistenceProvider(String mapName, String persistencePath) {
     this.mapName = mapName;
+    this.persistencePath = persistencePath;
     File dir = new File(getPersistencePath());
-    if (!dir.exists()) {
-      if (!dir.mkdir()) {
-        LOGGER.debug("Unable to create directory: {}", dir.getAbsolutePath());
-      }
+    if (!dir.exists() && !dir.mkdir()) {
```

Comment:
❗️Shouldn't we throw an exception if we fail instead of just logging?
❓What would happen if the directory gets created, nothing gets written to it, the system shuts down and this method gets called again when the system restarts? Would this prevent things from being persisted?

Reply:
You need to write an audit whenever you create a new file / directory or read an existing file / directory. You might also want to catch the AccessControlException that `dir.exists()` can throw and give a more helpful error message.